internal/schedule_client.go

```diff
@@ -363,6 +363,10 @@ type (
 		//
 		// [Visibility]: https://docs.temporal.io/visibility
 		TypedSearchAttributes SearchAttributes
+
+		// KeepOriginalWorkflowID - If true, and the action would start a workflow, a timestamp will not be
```

Comment:
Since this only applies if the action is `ScheduleWorkflowAction` is there a reason you didn't put the field on the action?

Reply:
Initially it was in ScheduleAction. Then @dnr pointed out that Policy is the more logical place for this field. There is no Policy in the SDK ScheduleOptions. So I left the field at the options top level. 
How can users set Policy from the workflow code?

Reply:
I am not sure why policy is more logical if this only applies to one type of action. @dnr do you expect this option to apply to other type of actions?

Reply:
We don't really have a distinct type for the start workflow action. Maybe we should have. I think "policies" make sense as general configuration that can affect schedule execution.

For other actions: suppose we had an action to start a top-level activity. I think top-level activities would still have names, so it could apply there too. Same for starting a nexus operation, though I'm not sure about naming there.

Reply:
hmm if this option is meant to apply to other future actions like top level activity of nexus operation then I think we should change the name to not include `Workflow` since that is specific to one type of action.